### DISCUSSION TITLE
dataset_class param in GeoDataModule

### DISCUSSION THREAD

**Discussion by roybenhayun** [torchgeo/torchgeo#1632] [Q&A]
see in answer for [another question](https://github.com/microsoft/torchgeo/discussions/1598):
> I would recommend extending GeoDataModule or NonGeoDataModule. That's what they're there for.

what should be the dataset_class param to GeoDataModule?
for example, in the [training-example](https://github.com/microsoft/satellite-imagery-labeling-tool/blob/main/example/training-example.ipynb), the module extends LightningDataModule and creates the dataset in the setup method. there is no class. 
how would that be refactored?

**Comment by roybenhayun** [comment]
just to note that the training-example is from few releases ago. wondering how this change will apply to either 0.4.1 and [0.5.0](https://github.com/microsoft/torchgeo/discussions/1520)

**Reply by adamjstewart** [reply]
From [the docs](https://torchgeo.readthedocs.io/en/stable/api/datamodules.html#torchgeo.datamodules.BaseDataModule.__init__):

> **dataset_class**  – Class used to instantiate a new dataset.

You can also see examples of this in https://github.com/microsoft/torchgeo/tree/main/torchgeo/datamodules

**Comment by roybenhayun** [comment]
tried to go over few of the builtin datamodules also in the context of building an ObjectDetectionTask. 
with ObjectDetectionTask specifically, there seems to be a conflict in some assertion.
some working code samples I have don't use datamodules. 

without some explaining post or a tutorial coupled with working examples it is not straightforward to understand from the builtin classes alone.
